dot/core/service.go

```diff
@@ -236,15 +226,10 @@ func (s *Service) handleBlock(block *types.Block, state *rtstorage.TrieState) er
 		return err
 	}
 
-	go func() {
-		s.Lock()
-		defer s.Unlock()
-		if s.ctx.Err() != nil {
-			return
-		}
-
-		s.blockAddCh <- block
-	}()
```

Comment:
We have a buffered block add channel (256), so we should not 'add to the queue' with goroutines, otherwise goroutines act as the 2nd queue when the channel is full. We should instead block to have proper back pressure.